sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -381,3 +381,76 @@ case class AlwaysFalse() extends Filter {
 @Evolving
 object AlwaysFalse extends AlwaysFalse {
 }
+
+/**
+ * Base class for collation aware string filters.
+ */
+abstract class CollatedFilter() extends Filter {
```

Comment:
to confirm: no data source supports these new filters yet, so no string collation filters will be pushed down, right?

Reply:
how about partition pruning? Which is fully controlled by Spark, not file source implementations.

Reply:
> to confirm: no data source supports these new filters yet, so no string collation filters will be pushed down, right?
Yes, that would be the idea here

Reply:
> We don't support partitioning on collated columns.

Oh I see, then we are safe.